### DISCUSSION TITLE
Getting CORS error when loading images on WASM

### DISCUSSION THREAD

**Discussion by LOOHP** [coil-kt/coil#2531] [Q&A] [answered]
I am trying to have an image displayed using compose on wasm. The image is being requested cross-site and I am getting a cors error.

```
Access to fetch at '...' from origin '...' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.
```

Is there any way to get around this? Since it is just showing an image and not like reading a resource for its contents.

**Comment by colinrtwhite** [comment, marked as answer]
It's not possible for Coil to work around the CORS security policy. Check out this thread [here](https://github.com/coil-kt/coil/issues/2448) for more info.

**Reply by LOOHP** [reply]
That's unfortunate, I thought since `<img>` in HTML doesn't have this issue there might be something that can be done. Thank you for your quick response nevertheless.

**Reply by colinrtwhite** [reply]
`<img>` is controlled by the browser and gets special privileges; the browser can guarantee that it's just downloading and rendering an image. Coil is user code that could technically be doing anything - including insecure operations.
